src/pages/index.js

```diff
@@ -45,40 +89,106 @@ addCardMOB.addEventListener("click", () => {
   addPlacePopup.open();
 });
 
+profilePic.addEventListener("click", () => {
+  profilePicPopup.open();
+});
+
 //***************************************************************************************************************************//
 //                                             Event Handlers                                                                //
 //***************************************************************************************************************************//
 
 function handleEditProfile(userData) {
+  const submitButton = editProfileModal.querySelector(
+    "#edit-profile-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+  api.patchProfileInfo(userData.name, userData.job);
   userInfo.setUserInfo(userData);
   editProfilePopup.popupForm.reset();
   editProfileFormValidator.toggleButtonState();
   editProfilePopup.close();
+  submitButton.textContent = "Save";
 }
 
 function handleAddCardSubmit(cardData) {
-  createCard(cardData, "prepend");
-  addPlacePopup.popupForm.reset();
-  addPlaceFormValidator.toggleButtonState();
-  addPlacePopup.close();
+  const submitButton = addCardModal.querySelector(
+    "#add-card-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardData)
+    .then((data) => {
+      addNewCard(data, "prepend");
+      addPlacePopup.popupForm.reset();
+      addPlaceFormValidator.toggleButtonState();
+      addPlacePopup.close();
+      submitButton.textContent = "Create";
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    });
+}
+
+let cardtoDeleteData;
+
+function handleConfirmDelete(data) {
+  confirmDeletePopup.open();
+  cardtoDeleteData = data;
+}
+
+function handleDeleteCard() {
+  api.deleteCard(cardtoDeleteData._id);
+  const image = document.querySelector(`img[src='${cardtoDeleteData.link}'`);
+  const deleteCard = image.closest(".card");
+  deleteCard.remove();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.